src/filename_inspector.py

```diff
@@ -0,0 +1,107 @@
+import fnmatch
+import glob
+import logging
+import os
+import json
+import csv
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def get_input(name: str) -> str:
+    value = os.getenv(f'INPUT_{name.upper()}')
+    return value
+
+
+def set_output(name: str, value: str, default_output_path: str = "default_output.txt"):
+    output_file = os.getenv('GITHUB_OUTPUT', default_output_path)
+    with open(output_file, 'a') as f:
+        f.write(f'{name}={value}\n')
+
+
+def set_failed(message: str):
+    logging.error(message)
+    exit(1)
+
+
+def _matches_name_patterns(file, name_patterns):
+    for name_pattern in name_patterns:
+        if fnmatch.fnmatch(file, name_pattern):
+            return True
+    return False
+
+
+def _matches_exclude(file, excludes):
+    for exclude in excludes:
+        if fnmatch.fnmatch(file, exclude):
+            return True
+    return False
+
+
+def find_non_matching_files(name_patterns, paths, excludes):
+    non_matching_files = []
+
+    # Iterate over each glob pattern to find files
+    for path in paths:
+        for file in glob.glob(path, recursive=True):
+            # Check if the file matches any of the exclude patterns
+            if _matches_exclude(file, excludes):
+                continue
+            # Check if the file matches any of the name patterns
+            if not _matches_name_patterns(file, name_patterns):
+                non_matching_files.append(file)
```

Comment:
```suggestion
            if any(re.match(p, file) for r in excludes):
                continue
            # Check if the file matches any of the name patterns
            if not any(re.match(p, file) for r in name_patterns):
                non_matching_files.append(file)
```

And you could remove 2 functions.

Reply:
Used in commit f2f7ad30a9323ebadb92d0ce3cd590837bc973d5.
Instead of re used fnmatch.